glide-core/redis-rs/redis/src/cluster_async/mod.rs

```diff
@@ -2337,6 +2453,58 @@ where
         }
     }
 
+    // The function is used to send an AUTH command to a node and then retry the original request.
+    // The function is used when the original request failed due to an AUTH error.
+    // Cases in which the function is helpful:
+    // 1. The password was changed, the connection was not re-established, and the request failed due to an AUTH error.
+    // 2. The protection method is allowing X hours of connection without authentication, and the request failed due to an AUTH error.
+    async fn send_re_auth_request(
+        core: Core<C>,
+        info: RequestInfo<C>,
+        address: String,
+    ) -> OperationResult {
+        let password = core
+            .get_cluster_param(|params| params.password.clone())
+            .expect(MUTEX_READ_ERR);
+        let username = core.get_cluster_param(|params| params.username.clone());
+        let Some(password) = password else {
+            return Err((
+                OperationTarget::Node { address },
+                RedisError::from((
+                    ErrorKind::AuthenticationFailed,
+                    "No password provided for AUTH",
```

Comment:
Can you keep the original error received by the server and chain it here? NOAUTH can have different error messages, so we're hiding information here , for example: https://github.com/valkey-io/valkey/blob/2df56d87c0ebe802f38e8922bb2ea1e4ca9cfa76/src/networking.c#L4119